File: Nx3Java/java/src/cx/EnumTools.java
package cx;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class EnumTools extends haxe.lang.HxObject
{
	public    EnumTools(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    EnumTools()
	{
		cx.EnumTools.__hx_ctor_cx_EnumTools(this);
	}
	
	
	public static   void __hx_ctor_cx_EnumTools(cx.EnumTools __temp_me22)
	{
		{
		}
		
	}
	
	
	public static  <T> T createFromString(java.lang.Class<T> e, java.lang.String str)
	{
		try 
		{
			java.lang.String type = str;
			haxe.root.Array<java.lang.String> params = new haxe.root.Array<java.lang.String>(new java.lang.String[]{});
			if (cx.StrTools.has(str, "(")) 
			{
				int parIdx = haxe.lang.StringExt.indexOf(str, "(", null);
				type = haxe.lang.StringExt.substr(str, 0, parIdx);
				params = haxe.lang.StringExt.split(haxe.root.StringTools.replace(haxe.root.StringTools.replace(haxe.lang.StringExt.substr(str, parIdx, null), "(", ""), ")", ""), ",");
			}
			
			return haxe.root.Type.createEnum(e, type, ((haxe.root.Array) (params) ));
		}
		catch (java.lang.Throwable __temp_catchallException315)
		{
			java.lang.Object __temp_catchall316 = __temp_catchallException315;
			if (( __temp_catchall316 instanceof haxe.lang.HaxeException )) 
			{
				__temp_catchall316 = ((haxe.lang.HaxeException) (__temp_catchallException315) ).obj;
			}
			
			{
				java.lang.Object e1 = __temp_catchall316;
				{
				}
				
			}
			
		}
		
		
		return null;
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new cx.EnumTools(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new cx.EnumTools();
	}
	
	
}

File: Nx3CS/bin/test/src/cx/EnumTools.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace cx{
	public  class EnumTools : global::haxe.lang.HxObject {
		public    EnumTools(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 8 "F:\\nx3\\src\\cx\\EnumTools.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    EnumTools(){
			unchecked {
				#line 8 "F:\\nx3\\src\\cx\\EnumTools.hx"
				global::cx.EnumTools.__hx_ctor_cx_EnumTools(this);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_cx_EnumTools(global::cx.EnumTools __temp_me24){
			unchecked {
				#line 8 "F:\\nx3\\src\\cx\\EnumTools.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public static   T createFromString<T>(global::System.Type e, string str){
			unchecked {
				#line 11 "F:\\nx3\\src\\cx\\EnumTools.hx"
				try {
					#line 12 "F:\\nx3\\src\\cx\\EnumTools.hx"
					string type = str;
					global::Array<object> @params = new global::Array<object>(new object[]{});
					if (global::cx.StrTools.has(str, "(")) {
						#line 15 "F:\\nx3\\src\\cx\\EnumTools.hx"
						int parIdx = global::haxe.lang.StringExt.indexOf(str, "(", default(global::haxe.lang.Null<int>));
						type = global::haxe.lang.StringExt.substr(str, 0, new global::haxe.lang.Null<int>(parIdx, true));
						@params = global::haxe.lang.StringExt.split(global::StringTools.replace(global::StringTools.replace(global::haxe.lang.StringExt.substr(str, parIdx, default(global::haxe.lang.Null<int>)), "(", ""), ")", ""), ",");
					}
					
					#line 19 "F:\\nx3\\src\\cx\\EnumTools.hx"
					return global::Type.createEnum<T>(e, type, @params);
				}
				catch (global::System.Exception __temp_catchallException178){
					#line 11 "F:\\nx3\\src\\cx\\EnumTools.hx"
					global::haxe.lang.Exceptions.exception = __temp_catchallException178;
					#line 20 "F:\\nx3\\src\\cx\\EnumTools.hx"
					object __temp_catchall179 = __temp_catchallException178;
					#line 20 "F:\\nx3\\src\\cx\\EnumTools.hx"
					if (( __temp_catchall179 is global::haxe.lang.HaxeException )) {
						#line 20 "F:\\nx3\\src\\cx\\EnumTools.hx"
						__temp_catchall179 = ((global::haxe.lang.HaxeException) (__temp_catchallException178) ).obj;
					}
					
					#line 20 "F:\\nx3\\src\\cx\\EnumTools.hx"
					{
						#line 20 "F:\\nx3\\src\\cx\\EnumTools.hx"
						object e1 = __temp_catchall179;
						#line 20 "F:\\nx3\\src\\cx\\EnumTools.hx"
						{
						}
						
					}
					
				}
				
				
				#line 21 "F:\\nx3\\src\\cx\\EnumTools.hx"
				return default(T);
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 8 "F:\\nx3\\src\\cx\\EnumTools.hx"
				return new global::cx.EnumTools(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 8 "F:\\nx3\\src\\cx\\EnumTools.hx"
				return new global::cx.EnumTools();
			}
			#line default
		}
		
		
	}
}


